src/nbl/ext/DebugDraw/CDraw3DLine.cpp

```diff
@@ -74,38 +74,34 @@ CDraw3DLine::CDraw3DLine(const core::smart_refctd_ptr<video::ILogicalDevice>& de
 	m_meshBuffer->setIndexCount(2);
 }
 
-void CDraw3DLine::recordToCommandBuffer(video::IGPUCommandBuffer* cmdBuffer, video::IGPUGraphicsPipeline* graphics_pipeline)
+void CDraw3DLine::recordToCommandBuffer(video::IGPUCommandBuffer* cmdBuffer, const IGPUBuffer* buffer, video::IGPUGraphicsPipeline* graphics_pipeline)
 {
 
 	auto cb = cmdBuffer;
 	assert(cb->getState() == IGPUCommandBuffer::ES_RECORDING);
-	auto* buf = m_meshBuffer->getVertexBufferBindings()->buffer.get();
 	size_t offset = 0u;
-	cb->bindVertexBuffers(0u, 1u, &buf, &offset);
+	cb->bindVertexBuffers(0u, 1u, &buffer, &offset);
 	cb->pushConstants(const_cast<nbl::video::IGPUPipelineLayout*>(m_meshBuffer->getPipeline()->getLayout()), asset::ISpecializedShader::ESS_VERTEX, 0, sizeof(m_viewProj), &m_viewProj);
 	cb->bindGraphicsPipeline(graphics_pipeline);
 	cb->draw(m_lines.size() * 2, 1u, 0u, 0u);
 
 }
 
-void CDraw3DLine::updateMeshBuffer()
+void CDraw3DLine::updateVertexBuffer(IGPUQueue* queue, core::smart_refctd_ptr<IGPUBuffer>& buffer)
```

Comment:
why do you take lines buffer from outside? Have it as a member of extension class, invisible to the user